src/adaptors/giveth/index.js

```diff
@@ -0,0 +1,380 @@
+const givDaiPairUnipoolContractInfo =
+  '0xa4523d703f663615bd41606b46b58deb2f926d98';
+
+// GIV - ETH Balancer
+const givEthBalancerPoolId =
+  '0x7819f1532c49388106f7762328c51ee70edd134c000200000000000000000109';
+const givEthBalancerAddress = '0xc0dbdca66a0636236fabe1b3c16b1bd4c84bb1e1';
+
+// ICHI Angel Vault
+const oneGivPairUnipoolContractInfo =
+  '0xa4b727df6fd608d1835e3440288c73fb28c4ef16';
+
+const defiLlamaGivId = `ethereum:${givTokenMainnetAddress}`;
+
+const tokenPairPoolQuery = gql`
+  {
+    pair(id: "<PLACEHOLDER>") {
+      id
+      reserveUSD
+      volumeUSD
+      reserve0
+      reserve1
+      totalSupply
+      token0 {
+        id
+        symbol
+      }
+      token1 {
+        id
+        symbol
+      }
+    }
+  }
+`;
+
+const balancerPairQuery = gql`
+  {
+    pool(id: "<PLACEHOLDER>") {
+      id
+      tokens {
+        id
+        address
+        balance
+        weight
+        symbol
+      }
+      totalShares
+    }
+  }
+`;
+
+const unipoolContractInfoQuery = gql`
+  {
+    unipool(id: "<PLACEHOLDER>") {
+      id
+      totalSupply
+      rewardRate
+    }
+  }
+`;
+
+const getPrices = async (tokens = []) => {
+  const idsSet = [defiLlamaGivId, ...tokens];
+  let prices = await utils.getData('https://coins.llama.fi/prices', {
+    coins: idsSet,
+  });
+  return prices.coins;
+};
+
+const balancerCalculatePoolApy = async (chainString, entry, contractInfo) => {
+  const givTokenAddress =
+    chainString === 'ethereum' ? givTokenMainnetAddress : givTokenGnosisAddress;
+
+  const weights = [entry.tokens[0].weight, entry.tokens[1].weight].map(
+    toBigNumberJs
+  );
+  const balances = [entry.tokens[0].balance, entry.tokens[1].balance].map(
+    toBigNumberJs
+  );
+
+  if (entry.tokens[0].address.toLowerCase() !== givTokenAddress) {
+    balances.reverse();
+    weights.reverse();
+  }
+
+  const lp = BigNumber(entry.totalShares)
+    .div(BigNumber.sum(...weights).div(weights[0]))
+    .div(balances[0]);
+
+  const totalSupply = BigNumber(contractInfo.totalSupply);
+  const apr = totalSupply.isZero()
+    ? null
+    : BigNumber(contractInfo.rewardRate)
+        .div(totalSupply)
+        .times(secsInOneYear)
+        .times('100')
+        .times(lp);
+  return apr;
+};
+
+const calculatePairApy = async (chainString, entry, contractInfo) => {
+  const givTokenAddress =
+    chainString === 'ethereum' ? givTokenMainnetAddress : givTokenGnosisAddress;
+  const tokenReserve = BigNumber(
+    entry.token0.id.toLowerCase() !== givTokenAddress
+      ? entry.reserve1
+      : entry.reserve0
+  );
+  const lp = BigNumber(entry.totalSupply)
+    .times(10 ** 18)
+    .div(2)
+    .div(tokenReserve);
+
+  const totalSupply = BigNumber(contractInfo.totalSupply);
+
+  const apr = totalSupply.isZero()
+    ? null
+    : BigNumber(contractInfo.rewardRate)
+        .div(totalSupply)
+        .times(secsInOneYear)
+        .times('100')
+        .times(lp)
+        .div(10 ** 18);
```

Comment:
What does this multiplication represent? Save it in a variable for clarity.

Reply:
I find it quite confusing the upscaling and downscaling as well. For instance, the `lp` variable is multiplied by 10e18, but right after, we divide `lp` by 10e18 to calculate APR. This only adds complexity to the math since the terms cancel each other. cc @mateodaza

Reply:
Thanks for noticing :)) the disadvantages of recklessly copy pasting haha